source/auth/tests/legacy/connection-string.yml

```diff
@@ -412,6 +412,28 @@ tests:
     mechanism_properties:
       ENVIRONMENT: azure
       TOKEN_RESOURCE: foo
+- description: should accept a url-encoded TOKEN_RESOURCE (MONGODB-OIDC)
+  uri: mongodb://user@localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:mongodb%3A//test-cluster
+  valid: true
+  credential:
+    username: user
+    password: null
+    source: $external
+    mechanism: MONGODB-OIDC
+    mechanism_properties:
+      ENVIRONMENT: azure
+      TOKEN_RESOURCE: mongodb%3A//test-cluster
+- description: should url-encode a TOKEN_RESOURCE (MONGODB-OIDC)
+  uri: mongodb://user@localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:a$b
+  valid: true
+  credential:
+    username: user
+    password: null
+    source: $external
+    mechanism: MONGODB-OIDC
+    mechanism_properties:
+      ENVIRONMENT: azure
+      TOKEN_RESOURCE: a%24b
```

Comment:
This should represent the unencoded raw value. For example, when the user sets the property via:
```
credential = credential.withMechanismProperty("TOKEN_RESOURCE", 
    credential.getMechanismProperty("TOKEN_RESOURCE"));
```
The value must not change.

The user should only have to url-encode the TOKEN_RESOURCE value when it is in the connection string, and contains certain reserved characters. If "$" is one of those characters, then the above uri is invalid, and we should throw an error (replace it with `,`, `&`, etc. to see why).

Whatever raw value is given, it of course must be url-encoded (just once, of course) when building the GCP or Azure URIs.